interpreter/cling/lib/Interpreter/IncrementalExecutor.cpp

```diff
@@ -240,9 +240,20 @@ void IncrementalExecutor::replaceSymbol(const char* Name, void* Addr) const {
 
 void* IncrementalExecutor::getAddressOfGlobal(llvm::StringRef symbolName,
                                               bool* fromJIT /*=0*/) const {
-  constexpr bool includeHostSymbols = true;
 
-  void* address = m_JIT->getSymbolAddress(symbolName, includeHostSymbols);
+  // Try the fast path first, going via the dlsym and the dynamic linker.
+  if (void* Addr = const_cast<void*>(platform::DLSym(symbolName.str()))) {
+    if (fromJIT)
+      *fromJIT = false;
+    return Addr;
+  }
+
+  // Try iterating over the registered llvm libraries.
+  if (void* Addr = llvm::sys::DynamicLibrary::SearchForAddressOfSymbol(symbolName.str())) {
+    if (fromJIT)
+      *fromJIT = false;
+    return Addr;
+  }
```

Comment:
This looks like a somewhat separate performance optimization that isn't explained in the commit message. Should this be its own commit?

Reply:
I've removed that from the PR for now - we need to rethink the `fromJIT` thingy anyways...